backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -226,6 +227,8 @@ private static org.sonarsource.sonarlint.core.rpc.protocol.common.Language toDto
   public GetAnalysisConfigResponse getAnalysisConfig(String configScopeId) {
     var bindingOpt = configurationRepository.getEffectiveBinding(configScopeId);
     var activeNodeJs = nodeJsService.getActiveNodeJs();
+    var inferredClientProperties = client.getInferredAnalysisProperties(new GetInferredAnalysisPropertiesParams(configScopeId)).join().getProperties();
+    clientAnalysisPropertiesRepository.setInferredProperties(configScopeId, inferredClientProperties);
```

Comment:
Do we need to store it in a repository if we ask for it everytime? Also the round trip on this line and the next one seems strange

Reply:
Ah I get it, it's because the repo has both user and inferred properties

Reply:
I've removed it and moved merge of the properties into `getAnalysisConfig()`. No need to store inferred properties in the repository.